phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -184,9 +184,24 @@ public boolean apply(PTable index) {
      */
     public static void serialize(PTable dataTable, ImmutableBytesWritable ptr, PhoenixConnection connection) {
         List<PTable> indexes = dataTable.getIndexes();
-        serialize(dataTable, ptr, indexes, connection);
+        serializeServerMaintainedIndexes(dataTable, ptr, indexes, connection);
     }
 
+    public static void serializeServerMaintainedIndexes(PTable dataTable, ImmutableBytesWritable ptr,
+            List<PTable> indexes, PhoenixConnection connection) {
+        Iterator<PTable> indexesItr = Collections.emptyListIterator();
+        boolean onlyLocalIndexes = dataTable.isImmutableRows() || dataTable.isTransactional();
+        if (onlyLocalIndexes) {
```

Comment:
Do we do anything special for transactional tables with local indexes?

Reply:
Long story short, we've disallowed local indexes on Omid for now due to PHOENIX-4760. Once this is fixed, we can turn it back on. Given that index updates are transactional with data updates,  there's not a great need IMHO for transactional local indexes, though. Having said that, there were a few things we needed to do for local indexes:
* Support client-side index maintenance for local indexes
* Handle the initial population (i.e. adding shadow columns) for a local index on the server side